lib/modules/manager/gomod/artifacts.ts

```diff
@@ -337,20 +337,23 @@ export async function updateArtifacts({
     await exec(execCommands, execOptions);
 
     const status = await getRepoStatus();
-    if (!status.modified.includes(sumFileName)) {
+    if (
+      !status.modified.includes(sumFileName) &&
+      !status.modified.includes(goModFileName)
+    ) {
       return null;
     }
 
-    logger.debug('Returning updated go.sum');
-    const res: UpdateArtifactsResult[] = [
-      {
+    const res: UpdateArtifactsResult[] = [];
+    if (status.modified.includes(sumFileName)) {
```

Comment:
```suggestion
    if (status.modified.includes(sumFileName)) {
      logger.debug('Returning updated go.sum');
```

Reply:
I though I included that. Sorry about that :)